cpp/include/cudf/rolling.hpp

```diff
@@ -57,6 +57,169 @@ std::unique_ptr<column> rolling_window(column_view const& input,
+ * The data would be grouped by `user_id`, and ordered by `day`-string. The aggregation 
+ * (SUM) would then be calculated for a window of 3 values around (and including) each row.
+ * 
+ * If the input rows were as follows:
+ * 
+ *  [ // user,  sales_amt
+ *    { "user1",   10      },
+ *    { "user2",   20      },
+ *    { "user1",   20      },
+ *    { "user1",   10      },
+ *    { "user2",   30      },
+ *    { "user2",   80      },
+ *    { "user1",   50      },
+ *    { "user1",   60      },
+ *    { "user2",   40      }
+ *  ]
+ * 
+ * Partitioning (grouping) by `user_id`, and ordering by `day` would yield the following `sales_amt` vector 
+ * (with 2 groups, one for each distinct `user_id`):
+ * 
+ *    [ 10,  20,  10,  50,  60,  20,  30,  80,  40 ]
+ *      <-------user1-------->|<------user2------->
+ * 
+ * The SUM aggregation would then be applied, with 1 preceding, and 1 following
+ * row, with a minimum of 1 period. The aggregation window is thus 3 rows wide,
+ * thereby yielding the following column:
+ * 
+ *    [ 30, 40,  80, 120, 100,  50, 130, 150, 120 ]
+ * 
+ * Note: The SUMs calculated at the group boundaries (i.e. indices 0, 4, 5, and 8)
+ * consider only 2 values each, in spite of the window-size being 3.
+ * Each aggregation operation cannot cross group boundaries.
+ * 
+ * The returned column for `op == COUNT` always has `INT32` type. All other operators return a 
+ * column of the same type as the input. Therefore it is suggested to convert integer column types
+ * (especially low-precision integers) to `FLOAT32` or `FLOAT64` before doing a rolling `MEAN`.
+ *
+ * @param[in] grouping_keys The (pre-sorted) grouping columns
+ * @param[in] input The input column (to be aggregated)
+ * @param[in] preceding_window The static rolling window size in the backward direction.
+ * @param[in] following_window The static rolling window size in the forward direction.
+ * @param[in] min_periods Minimum number of observations in window required to have a value,
+ *                        otherwise element `i` is null.
+ * @param[in] aggr The rolling window aggregation type (SUM, MAX, MIN, etc.)
+ *
+ * @returns   A nullable output column containing the rolling window results
+ **/
+std::unique_ptr<column> grouped_rolling_window(table_view const& grouping_keys,
+                                               column_view const& input,
+                                               size_type preceding_window,
+                                               size_type following_window,
+                                               size_type min_periods,
+                                               std::unique_ptr<aggregation> const& aggr,
+                                               rmm::mr::device_memory_resource* mr = 
+                                                 rmm::mr::get_default_resource());
+
+/**
+ * @brief  Applies a grouping-aware, timestamp-based rolling window function to the values in a column.
```

Comment:
See above edit suggestions.

Reply:
Accepted all of the above.

Reply:
Right, I just mean you need to duplicate the suggestions for this comment block...

Reply:
Yikes, will do.